catalog/core/catalog-core-standardframework/src/main/java/ddf/catalog/impl/operations/ResourceOperations.java

```diff
@@ -552,139 +550,126 @@ protected ResourceInfo getResourceInfo(ResourceRequest resourceRequest, String s
-                                    + value.getClass()
-                                    + "' is not supported by this instance of the CatalogFramework.");
-                }
-            } else {
-                throw new ResourceNotSupportedException(
-                        "The GetResourceRequest with attribute name '" + name
-                                + "' is not supported by this instance of the CatalogFramework.");
+                String metacardId = (String) value;
+                LOGGER.debug("metacardId = {},   site = {}", metacardId, site);
+                query = createMetacardIdQuery(metacardId);
             }
+
+            QueryRequest queryRequest = new QueryRequestImpl(anyTag(query,
+                    site,
+                    isEnterprise),
+                    isEnterprise,
+                    Collections.singletonList(site == null ? this.getId() : site),
+                    resourceRequest.getProperties());
+
+            resourceInfo = getResourceInfo(queryRequest,
+                    resourceUri,
+                    requestProperties,
+                    federatedSite,
+                    fanoutEnabled);
         } catch (UnsupportedQueryException | FederationException e) {
 
             throw new ResourceNotFoundException(DEFAULT_RESOURCE_NOT_FOUND_MESSAGE, e);
         }
 
-        LOGGER.debug("Returning resourceURI: {}", resourceUri);
-        if (resourceUri == null) {
+        if (resourceInfo.getResourceUri() == null) {
             throw new ResourceNotFoundException(DEFAULT_RESOURCE_NOT_FOUND_MESSAGE);
         }
+        LOGGER.debug("Returning resourceURI: {}", resourceInfo.getResourceUri());
+        return resourceInfo;
+    }
+
+    private ResourceInfo getResourceInfo(QueryRequest queryRequest, URI uri,
+            Map<String, Serializable> requestProperties, StringBuilder federatedSite,
+            boolean fanoutEnabled)
+            throws ResourceNotFoundException, UnsupportedQueryException, FederationException {
+        Metacard metacard;
+        URI resourceUri = uri;
+        QueryResponse queryResponse = queryOperations.query(queryRequest,
+                null,
+                true,
+                fanoutEnabled);
+        if (queryResponse.getResults()
+                .isEmpty()) {
+            throw new ResourceNotFoundException(
+                    "Could not resolve source id for URI by doing a URI based query.");
+        }
+        metacard = queryResponse.getResults()
+                .get(0)
+                .getMetacard();
+        if (uri != null && queryResponse.getResults()
+                .size() > 1) {
+            for (Result result : queryResponse.getResults()) {
+                if (uri.equals(result.getMetacard()
+                        .getResourceURI())) {
+                    metacard = result.getMetacard();
+                    break;
+                }
+            }
+        }
+        if(resourceUri == null) {
```

Comment:
✏️  you could consider moving this up since `resourceUri` gets initialized on line 621. But doesn't really add any value.